packages/grid/x-data-grid-pro/src/components/DataGridProVirtualScroller.tsx

```diff
@@ -443,6 +441,7 @@ const DataGridProVirtualScroller = React.forwardRef<
         </VirtualScrollerPinnedRows>
       ) : null}
       <GridVirtualScrollerContent {...contentProps}>
+        <GridOverlays />
```

Comment:
This change made the loading overlay move its position on scroll 😅

https://github.com/mui/mui-x/assets/13808724/dc4c4d4b-3c21-47af-bb2c-4aa55c0fdfd0

Reply:
mm a bit of investigation - this one would be tricky not only because of the position of the overlay but because if it is `loading` then the overlay needs to cover all the rows (including the pinned ones)

Reply:
Makes sense, I think leaving the overlay where it was is better then.

Reply:
I opted for a 3rd approach - keep the newly created slot and create an additional component `GridInnerOverlay` that is added deeper in the DOM. The result is no breaking changes to existing functionality and still the issue with the pinned columns is resolved.
If this is not an option then we are back to the simple fix of `z-index: 5` 😅

Reply:
Well the simple fix does seem nice in light of this information :sweat_smile: so I would suggest to go back to that at this point.